src/libraries/System.Transactions.Local/src/System/Transactions/TransactionManager.cs

```diff
@@ -391,6 +399,56 @@ public static TimeSpan MaximumTimeout
             }
         }
 
+        /// <summary>
+        /// Controls whether usage of System.Transactions APIs that require escalation to a distributed transaction will do so;
+        /// if your application requires distributed transaction, opt into using them by setting this to <see langword="true" />.
+        /// If set to <see langword="false" /> (the default), escalation to a distributed transaction will throw a <see cref="NotSupportedException" />.
+        /// </summary>
+#if WINDOWS
+        public static bool ImplicitDistributedTransactions
+        {
+            get => DtcProxyShimFactory.s_transactionConnector is not null;
+
+            [SupportedOSPlatform("windows")]
+            [RequiresUnreferencedCode(DistributedTransactionTrimmingWarning)]
+            set
+            {
+                // Make sure this flag can only be set once, and that once distributed transactions have been initialized,
+                // it's frozen.
+                if (s_implicitDistributedTransactions is null)
+                {
+                    s_implicitDistributedTransactions = value;
+                }
+                else if (value != s_implicitDistributedTransactions)
+                {
+                    throw new InvalidOperationException(SR.ImplicitDistributedTransactionsCannotBeChanged);
+                }
+
+                if (value)
+                {
+                    DtcProxyShimFactory.s_transactionConnector ??= new DtcProxyShimFactory.DtcTransactionConnector();
+                }
```

Comment:
Do we need to make this setter thread-safe?  At the moment two threads racing to call set could hit issues, like double-allocating the DtcTransactionConnector.

Reply:
This is a class with no fields, made just to allow trimming the distributed transactions code if we don't set the flag. I don't think we should bother.

Reply:
That's not the only issue.  Two threads could be racing to set this, one with true, one with false.  What behavior do you expect, and how does this implementation guarantee that?